docs/managing-templates.md

```diff
@@ -134,3 +134,154 @@ Now let's make use of our new template.
         }
     ]
     ```
+
+## Adding a required field
+
+In may cases, we want users of the template to provide more detail, such as
+data that can be used to tailor the notifications based on the target at
+hand. This is accomplished via a [form fields](declarative-templates.md#example-form-fields).
+
+Let's make a new template that enables notifications via [Azure Devops Work
+Items](notifications/ado.md), where we require the user to specify the Area
+Path that the work items OneFuzz will create.
+
+This example will demonstrate setting the following:
+* `Project` via the project name specified during job creation.
+* `Area Path` via the new required field.
+* `Iteration Path` via a new optional field, with a default value in the template.
+
+1. Enable support for declarative templates.
+    ```
+    onefuzz config --enable_feature job_templates
+    ```
+2. Save a copy of the template locally.
+    ```
+    onefuzz job_templates manage get libfuzzer_linux > libfuzzer_linux_ado_areapath.json
+    ```
+3. With your preferred text editor, add the following to the `notifications` list:
+    ```json 
+    {
+      "config": {
+        "base_url": "https://dev.azure.com/org_name",
+        "auth_token": "ADO_AUTH_TOKEN",
+        "type": "Bug",
+        "project": "{{ job.project }}",
+        "ado_fields": {
+          "System.AreaPath": "{{ task.tags['area_path'] }}",
+          "Microsoft.VSTS.Scheduling.StoryPoints": "1",
+          "System.IterationPath": "{{ task.tags['iteration_path'] }}",
+          "System.Title": "{{ report.crash_site }} - {{ report.executable }}",
+          "Microsoft.VSTS.TCM.ReproSteps": "This is my call stack: <ul> {% for item in report.call_stack %} <li> {{ item }} </li> {% endfor %} </ul>"
+        },
+        "comment": "This is my comment. {{ report.input_sha256 }} {{ input_url }} <br> <pre>{{ repro_cmd }}</pre>",
+        "unique_fields": ["System.Title", "System.AreaPath"],
+        "on_duplicate": {
+          "comment": "Another <a href='{{ input_url }}'>POC</a> was found in <a href='{{ target_url }}'>target</a>. <br> <pre>{{ repro_cmd }}</pre>",
+          "set_state": { "Resolved": "Active" },
+          "ado_fields": {
+            "System.IterationPath": "{{ task.tags['iteration_path'] }}"
+          },
+          "increment": ["Microsoft.VSTS.Scheduling.StoryPoints"]
+        }
+      }
+    }
+    ```
+4. With your preferred text editor, add the following to user fields to the end of the `user_fields` list.
+    1. A required field that specifies the tag name `area_path`.
+        ```json
+        {
+            "help": "Area path for reported crashes",
+            "locations": [
+                {
+                    "op": "add",
+                    "path": "/tasks/1/tags/area_path"
+                }
+            ],
+            "name": "area_path",
+            "required": true,
+            "type": "Str"
+        }
+        ```
+    2. An optional field that specifies the `iteration_path`
```

Comment:
Again, consider using something like `my_iteration_path` to emphasize that it is non-generic, but an example of some concrete value.

Reply:
Disregard this suggestion.